src/crepe/api/Scene.h

```diff
@@ -7,19 +7,33 @@ namespace crepe {
 class SceneManager;
 class ComponentManager;
 
+/**
+ * \brief Represents a Scene
+ * 
+ * This class represents a Scene. The Scene class is only used as an interface for the game
+ * programmer.
+ */
 class Scene {
 protected:
+	/**
+	 * \param mgr  Reference to the ComponentManager
+	 * \param name  Name of the scene
+	 */
 	Scene(ComponentManager & mgr, const std::string & name);
+	//! SceneManager instances Scene
 	friend class SceneManager;
 
 public:
 	virtual ~Scene() = default;
 
 public:
+	//! Load the scene
 	virtual void load_scene() = 0;
+	//! The scene name
```

Comment:
Ik snap alleen niet waarom `name` niet een property is die de concrete scène zelf definiëert (net zoals `get_instances_max` bij `Component`)? Nu geef je de scène pas een naam bij het registereren aan de `SceneManager`, maar ik zou verwachten dat dit exact één keer gebeurt per concrete scène, en dat het wisselen van scènes daarna gewoon met de `std::type_index` van de concrete klasse gaat?

Reply:
Je hebt gelijk dat het wellicht iets minder intuïtief is om `name` te setten zodra je de scene registreert. Ik zal dit aanpassen in deze PR, maar eigenlijk was deze PR alleen voor Doxygen comments (en kwam ik er toevallig achter dat bij Scene en SceneManager nog Doxygen comments missen).

Reply:
Ik ga het niet veranderen dat je d.m.v. `std::type_index` een scene kunt wisselen, mij lijkt het als game programmeur veel handiger om dat d.m.v. de `name` te doen.